### DISCUSSION TITLE
multi-stage search

### DISCUSSION THREAD

**Discussion by imaglib** [milvus-io/milvus#3873] [Q&A and General discussion] [answered]
I want do multi-stage search, for example ,1. first stage is binary hamming and second stage is image embeding .2 first stage is word embeding and second stage is image embeing . Further more user can specify rules of multi-stage indexing.

**Comment by shiyu22** [comment, marked as answer]
Are you saying the multi-stage search is [multi-attribute search](https://github.com/milvus-io/milvus/issues/3603) like the issue?

If you just want multiple embeddings to be retrieved, you can create multiple collections to store different embeddings data.

**Reply by imaglib** [reply]
Do you have example of multi-attribute search , like insert data , build index and search ?

**Reply by imaglib** [reply]
I want to use two vectors to search, the first stage is vec1 and use vec2 to rerank condidates of stage1 .

**Reply by shiyu22** [reply]
As you can see, multi-attribute search is a new feature we're planning, so we don't have an example of that, and stay tuned.

**Reply by imaglib** [reply]
When will it be ready?

**Reply by shiyu22** [reply]
About the progress, you can follow this [issue](https://github.com/milvus-io/milvus/issues/3603).
